federation-2/router-bridge/src/planner.rs

```diff
@@ -147,30 +148,101 @@ pub struct PlannerSetupError {
     pub stack: Option<String>,
 }
 
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "camelCase")]
+/// A list of fields that will be resolved
+/// for a given type
+pub struct ReferencedFieldsForType {
+    /// names of the fields queried
+    pub field_names: Option<Vec<String>>,
+    /// whether the field is an interface
+    pub is_interface: Option<bool>,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "camelCase")]
+/// UsageReporting fields, that will be used
+/// to send stats to uplink/studio
+pub struct UsageReporting {
+    /// this is a query related hash that will allow uplink
+    /// to bucket the stats correctly
+    pub stats_report_key: String,
+    /// a list of all types and fields referenced in the query
+    pub referenced_fields_by_type: IndexMap<String, ReferencedFieldsForType>,
+}
+
 #[derive(Deserialize, Debug)]
-/// The result of a router bridge invocation
+#[serde(rename_all = "camelCase")]
+/// The result of a router bridge plan_worker invocation
 pub struct PlanResult<T> {
     /// The data if the query was successfully run
     pub data: Option<T>,
+    /// Usage reporting related data such as the
+    /// operation signature and referenced fields
+    pub usage_reporting: Option<UsageReporting>,
     /// The errors if the query failed
     pub errors: Option<Vec<PlanError>>,
 }
 
+/// The payload if the plan_worker invocation succeeded
+#[derive(Debug)]
+pub struct PlanSuccess<T> {
+    /// The payload you're looking for
+    pub data: T,
+    /// Usage reporting related data such as the
+    /// operation signature and referenced fields
+    pub usage_reporting: Option<UsageReporting>,
+}
+
+/// The payload if the plan_worker invocation failed
+#[derive(Debug, Clone)]
+pub struct PlanErrors {
+    /// The errors the plan_worker invocation failed with
+    pub errors: Arc<Vec<PlanError>>,
```

Comment:
why is there an Arc here?

Reply:
It's because routerservice boxerrors need Clone, i totally agree this probably doesnt belong here tho, shall i change it ?